addons/hr_holidays/models/hr_holidays.py

```diff
@@ -455,44 +503,30 @@ def holidays_validate(self):
                     }
                     leaves += self.with_context(mail_notify_force_send=False).create(values)
                 # TODO is it necessary to interleave the calls?
-                for signal in ('confirm', 'validate', 'second_validate'):
-                    leaves.signal_workflow(signal)
+                leaves.action_approve()
+                if leaves[0].double_validation:
```

Comment:
Leaves is a recordset of leave requests, we need to check if double_validation is activated, we do this by checking the double_validation field of any of those leaves (i.e. [0] here). 
The leaves are all created in the for loop just before, so they all have the same value for the double_validation field.